plugins/modules/purefa_pod_replica.py

```diff
@@ -85,133 +92,165 @@
 RETURN = """
 """
 
-MIN_REQUIRED_API_VERSION = "1.19"
+CONTEXT_VERSION = "2.38"
 
 from ansible.module_utils.basic import AnsibleModule
 from ansible_collections.purestorage.flasharray.plugins.module_utils.purefa import (
-    get_system,
+    get_array,
     purefa_argument_spec,
 )
+from ansible_collections.purestorage.flasharray.plugins.module_utils.version import (
+    LooseVersion,
+)
 
 
 def get_local_pod(module, array):
     """Return Pod or None"""
-    try:
-        return array.get_pod(module.params["name"])
-    except Exception:
+    api_version = array.get_rest_version()
+    if LooseVersion(CONTEXT_VERSION) <= LooseVersion(api_version):
+        res = array.get_pods(
+            names=[module.params["name"]], context_names=[module.params["context"]]
+        )
+    else:
+        res = array.get_pods(names=[module.params["name"]])
+    if res.status_code != 200:
         return None
+    else:
+        return list(res.items)[0]
 
 
 def get_local_rl(module, array):
     """Return Pod Replica Link or None"""
-    try:
-        rlinks = array.list_pod_replica_links()
-        for link in range(0, len(rlinks)):
-            if rlinks[link]["local_pod_name"] == module.params["name"]:
-                return rlinks[link]
-        return None
-    except Exception:
+    api_version = array.get_rest_version()
+    if LooseVersion(CONTEXT_VERSION) <= LooseVersion(api_version):
+        res = array.get_pod_replica_links(
+            local_pod_names=[module.params["name"]],
+            context_names=[module.params["context"]],
+            total_item_count=True,
+        )
+    else:
+        res = array.get_pod_replica_links(
+            local_pod_names=[module.params["name"]], total_item_count=True
+        )
+    if res.total_item_count == 0:
         return None
-
-
-def _get_arrays(array):
-    """Get Connected Arrays"""
-    arrays = []
-    array_details = array.list_array_connections()
-    for arraycnt in range(0, len(array_details)):
-        arrays.append(array_details[arraycnt]["array_name"])
-    return arrays
+    else:
+        return list(res.items)[0]
 
 
 def update_rl(module, array, local_rl):
     """Create Pod Replica Link"""
+    api_version = array.get_rest_version()
     changed = False
     if module.params["pause"] is not None:
-        if local_rl["status"] != "paused" and module.params["pause"]:
+        if local_rl.status != "paused" and module.params["pause"]:
             changed = True
             if not module.check_mode:
-                try:
-                    array.pause_pod_replica_link(
+                if LooseVersion(CONTEXT_VERSION) <= LooseVersion(api_version):
+                    res = array.pause_pod_replica_link(
```

Comment:
array doesn't have pause_pod_replica_link attribute, should use "patch_pod_replica_links"